src/transformers/models/wav2vec2_with_lm/processing_wav2vec2_with_lm.py

```diff
@@ -70,15 +70,15 @@ class Wav2Vec2ProcessorWithLM(ProcessorMixin):
     with language model support into a single processor for language model boosted speech recognition decoding.
 
     Args:
-        feature_extractor ([`Wav2Vec2FeatureExtractor`]):
-            An instance of [`Wav2Vec2FeatureExtractor`]. The feature extractor is a required input.
+        feature_extractor ([`AutoFeatureExtractor`]):
+            An instance of [`AutoFeatureExtractor`]. The feature extractor is a required input.
         tokenizer ([`Wav2Vec2CTCTokenizer`]):
             An instance of [`Wav2Vec2CTCTokenizer`]. The tokenizer is a required input.
         decoder (`pyctcdecode.BeamSearchDecoderCTC`):
             An instance of [`pyctcdecode.BeamSearchDecoderCTC`]. The decoder is a required input.
     """
 
-    feature_extractor_class = "Wav2Vec2FeatureExtractor"
+    feature_extractor_class = "AutoFeatureExtractor"
```

Comment:
Perhaps this is less vulnerable to unexpected behaviour? Or do you think we should accept all feature extractors as valid attributes of the Wav2Vec2 processor?
```suggestion
    feature_extractor_class = ("Wav2Vec2FeatureExtractor", "SeamlessM4TFeatureExtractor")
```

Reply:
Because of the tokenizer behaviour, when passing a tuple to `..._class`, the second class is chosen by default if `use_fast=True` (and the other way around).  So in that case, it won't work.

This is the first time that we had to face a processor with two possible classes. I'd be happy to modify the `ProcessorMixin` behaviour but I believe passing `AutoFeatureExtractor` is an easier workaround, especially since I don't really see a case in which an user would pass another FE yet!

cc @amyeroberts and @ArthurZucker, what do you think of this ?

Reply:
I see - good with me to use `AutoFeatureExtractor` in this case since changing the `ProcessorMixin` would add unnecessary complexity to handle this unique example

Reply:
This isn't the first processor to handle the case with two object -- that's Instruct BLIP -- although it's slightly different as that has two tokenizers, rather than accepts two tokenizers. 

I'm not a fan of using `AutoFeatureExtractor` here as the processor doesn't accept any feature extractor, and so the type is misleading. Ideally the processor should be adapted so for e.g. tokenizers it can accept a list of lists e.g. `[[ToknizerA, TokenizerAFast], [TokenizerB, TokenizerBFast]]`, and to accept lists of objects for the other types e.g. `[FeatureExtractorA, FeatureExtractorB, FeatureExtractorC]`. 

At the very least, the processor doc should indicate that only these two type are accepted and type verification happens in the init.

Reply:
I am hitting the same issue at #29414, however I don't really see the problem of using autoclasses in the processors, I just would like it to be consistent (use either both auto classes for feature extractor and tokenizer, or not at all).

Cause this PR currently would use AutoFeatureExtractor for the feature extractor, but a specific tokenizer class for the tokenizer. Hence I've opened #29414 to make the discussion more general, feel free to comment there!